utam-compiler/src/main/java/utam/compiler/helpers/TypeUtilities.java

```diff
@@ -192,12 +254,21 @@ public Class getClassType() {
 
   public static class Element implements TypeProvider {
     private final String name;
-    private final String containingType;
     private final List<BasicElementInterface> basicInterfaces = new ArrayList<>();
 
-    Element(String name, String[] interfaceTypes, String containingType) {
-      this.name = name.substring(0, 1).toUpperCase() + name.substring(1) + "Element";
-      this.containingType = containingType;
+    Element(String name, String[] interfaceTypes, boolean requiresPrefix) {
+      // If the basic element being defined is in an implementation-only Page Object,
+      // that is, one that has an "implements" property defined, and the element is
+      // marked as public, then there must be a method defined in the interface-only
+      // Page Object that matches the name of the exposed element. Assuming the method
+      // is named "getFoo," that method will have a generated return type of interface
+      // "GetFooElement", and the implementation Page Object must have an implementation
```

Comment:
to better understand it, would love to see couple JSON files in unit tests as an example.

Reply:
Added in next commit.